### PR TITLE
Update DG with implementation details for loan view GUI

### PR DESCRIPTION
**Pull request by marcus-ny**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -302,10 +302,12 @@ _{more aspects and alternatives to be added}
 ### Loan Analytics - Joseph
 
 #### Implementation
+
```
Comment (Joseph31416) [inline docs/DeveloperGuide.md:305]: Good adherence to checkstyle.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -365,23 +370,60 @@ Sequence diagram for the deletion of a loan:
...
+To ensure that only either the loan list or the person list is displayed, an additional `BooleanProperty` is added to
+the `Model`
+component to act as a flag to indicate which list is currently being displayed. This flag is updated by corresponding
+commands.
+For instance, commands such as `list` will toggle the flag to false, while `viewloan` will toggle the flag to true.
+This update switches the display between the two lists inside `MainWindow`.
+
+#### Design Considerations
+
+##### Aspect: How the GUI is updated
```
Comment (Joseph31416) [inline docs/DeveloperGuide.md:417]: Good consideration of implementation.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -365,23 +370,60 @@ Sequence diagram for the deletion of a loan:
...
     * Pros: More centralized command parsing.
-    * Cons: May result in the `AddressBookParser` class becoming too large. This also goes against various SWE principles
-  ,and makes the code harder to maintain.
+    * Cons: May result in the `AddressBookParser` class becoming too large. This also goes against various SWE
+      principles
+      ,and makes the code harder to maintain.
+
+### Loan view GUI - Kyal Sin Min Thet
+
+#### Implementation
```
Comment (Joseph31416) [inline docs/DeveloperGuide.md:397]: Clear description of implementation.

**Review by Joseph31416 — APPROVED** [review]
Overall this looks good to me.

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -365,23 +370,60 @@ Sequence diagram for the deletion of a loan:
...
+
+#### Design Considerations
+
+##### Aspect: How the GUI is updated
+
+* **Alternative 1 (current choice):** The GUI updates are done by the `Model` component's observable properties.
+    * Pros: Follows the observer design pattern, reducing coupling between the `Model` and `MainWindow` components.
+    * Cons: The GUI updates are restricted to the observable properties of the `Model` component.
+
+* **Alternative 2:** The GUI updates are done by the `MainWindow` component.
```
Comment (xiaorui-ui) [inline docs/DeveloperGuide.md:423]: Realistic alternative considered

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -402,6 +444,7 @@ Sequence diagram for the deletion of a loan:
 **Target user profile**:
 
 The target user is businessman who satisfies the following criteria
+
```
Comment (xiaorui-ui) [inline docs/DeveloperGuide.md:447]: Good adherance to checksyle

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -365,23 +370,60 @@ Sequence diagram for the deletion of a loan:
...
+##### Aspect: How the GUI is updated
+
+* **Alternative 1 (current choice):** The GUI updates are done by the `Model` component's observable properties.
+    * Pros: Follows the observer design pattern, reducing coupling between the `Model` and `MainWindow` components.
+    * Cons: The GUI updates are restricted to the observable properties of the `Model` component.
+
+* **Alternative 2:** The GUI updates are done by the `MainWindow` component.
+    * Pros: More explicit control over the GUI updates.
+    * Cons: `Model` needs a reference to `MainWindow` to update the GUI directly. This increases coupling between the
+      components.
```
Comment (xiaorui-ui) [inline docs/DeveloperGuide.md:426]: Careful analysis of pros and cons, good!

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -604,34 +644,35 @@ Use case ends.
...
 * **Actor**: A person or thing that performs an action.
 * **MSS**: Main Success Scenario, the most common path through a use case.
 * **Extensions**: The alternative paths through a use case.
-* **Non-Functional Requirements**: A requirement that specifies criteria that can be used to judge the operation of 
-a system, rather than specific behaviours.
+* **Non-Functional Requirements**: A requirement that specifies criteria that can be used to judge the operation of
+  a system, rather than specific behaviours.
 * **Mainstream OS**: Windows, Linux, Unix, or MacOS
 
 --------------------------------------------------------------------------------------------------------------------
```
Comment (xiaorui-ui) [inline docs/DeveloperGuide.md:678]: lgtm

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -319,7 +321,9 @@ It contains the following fields that can prove to be useful for the user:
 ![AnalyticsSequenceDiagram](images/AnalyticsSequenceDiagram.png)
 
 #### Design considerations:
+
 ##### Aspect: Initialization of the Analytics object:
+
```
Comment (kjw142857) [inline docs/DeveloperGuide.md:326]: I like how you left blank lines to increase clarity and structure of the file!

File: docs/DeveloperGuide.md [hunk 9]
```diff
@@ -347,13 +351,14 @@ parsed and transformed into an appropriate format.
 The parsing of the command is done by the `DeleteLoanCommandParser` class, which is responsible for parsing the user
 input.
 
-The `DeleteLoanCommand` class is instantiated in the `DeleteLoanCommandParser` class, while the 
+The `DeleteLoanCommand` class is instantiated in the `DeleteLoanCommandParser` class, while the
```
Comment (kjw142857) [inline docs/DeveloperGuide.md:354]: Nice catch on the removed whitespace!

File: docs/DeveloperGuide.md [hunk 10]
```diff
@@ -365,23 +370,60 @@ Sequence diagram for the deletion of a loan:
...
 ##### Aspect: How the command is executed:
+
 * **Alternative 1 (current choice):** The `DeleteLoanCommand` class is responsible for executing the command only.
     * Pros: Follows the Single Responsibility Principle. Simpler to debug.
     * Cons: May result in more classes.
 * **Alternative 2:** The `LogicManager` class is responsible for executing the command.
     * Pros: More centralized command execution.
     * Cons: May result in the `LogicManager` class becoming too large. This also goes against various SWE principles,
-  and makes the code harder to maintain.
+      and makes the code harder to maintain.
```
Comment (kjw142857) [inline docs/DeveloperGuide.md:382]: Again, good effort to standardise the formatting!

File: docs/DeveloperGuide.md [hunk 11]
```diff
@@ -365,23 +370,60 @@ Sequence diagram for the deletion of a loan:
...
+The GUI component to display loans attached to a contact is implemented in tandem with the `viewloan` command.
+
+The update behaviour is achieved through the use of `ObservableList` objects in the `Model` component. The `MainWindow`
+component listens for changes in the `ObservableList` objects and updates the GUI accordingly.
+
+The `LoanListPanel` (similar to `PersonListPanel`) is responsible for displaying the list of loans attached to a
+contact. It generates new `LoanCard` objects according to the `loanList` in the `Model` class.
+To accommodate the new GUI component, the `MainWindow.java` file is updated to include the new `LoanListPanel`.
+
+To ensure that only either the loan list or the person list is displayed, an additional `BooleanProperty` is added to
```
Comment (kjw142857) [inline docs/DeveloperGuide.md:408]: Nice explanation of why the `BooleanProperty` is abstracted out!

File: docs/DeveloperGuide.md [hunk 12]
```diff
@@ -365,23 +370,60 @@ Sequence diagram for the deletion of a loan:
...
+the `Model`
+component to act as a flag to indicate which list is currently being displayed. This flag is updated by corresponding
+commands.
+For instance, commands such as `list` will toggle the flag to false, while `viewloan` will toggle the flag to true.
```
Comment (kjw142857) [inline docs/DeveloperGuide.md:412]: Clear and concise examples of what will cause the flag to toggle, well done!

**Review by kjw142857 — APPROVED** [review]
LGTM!

**Pull request merged**